tools/cold-backup/run.sh

```diff
@@ -0,0 +1,252 @@
+                .metadata.selfLink, .items[].metadata.selfLink, \
+                .metadata.uid, .items[].metadata.uid, \
+                .status, .items[].status)"
+            local id=$(printf $name | sed 's/\//-/g;s/ //g')
+            oc get $crs -o yaml | yq eval "$exp" - > $__TMP/resources/$SOURCE_NS-$id.yaml
+        fi
+    fi
+}
+
+__create_pvc() {
+    local pvc="$1"
+    local size="$2"
+    local class="${3-}"
+    if [[ -n $pvc && -n $size ]]; then
+        local exp="s/\$pvc/$pvc/g; s/\$size/$size/g; /storageClassName/d"
+        if [[ -n $class ]]; then
+            exp="s/\$pvc/$pvc/g; s/\$size/$size/g; s/\$class/$class/g"
+        fi
+        echo "Creating pvc $pvc of size $size"
+        sed "$exp" $__HOME/pvc.yaml | oc create -f -
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+__rsync() {
+    local pvc="$1"
+    local from="$2"
+    local to="$3"
+    if [[ -n $pvc && -n $from && -n $to ]]; then
+        local patch=$(sed "s/\$pvc/$pvc/g" $__HOME/patch.json)
+        oc run $__MPOD --image="dummy" --restart="Never" --overrides="$patch"
+        oc wait --for condition="ready" pod $__MPOD --timeout="300s"
+        oc rsync --no-perms --delete --progress $from $to
+        oc delete pod $__MPOD
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+__compress() {
+    if [[ -n $BACKUP_NAME && -n $BACKUP_HOME ]]; then
+        echo "Compressing"
+        local current_dir=$(pwd)
+        cd $__TMP
+        zip -qr $BACKUP_NAME.zip *
+        mv $BACKUP_NAME.zip $BACKUP_HOME
+        cd $current_dir
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+__uncompress() {
+    if [[ -n $BACKUP_NAME && -n $BACKUP_HOME ]]; then
+        echo "Uncompressing"
+        rm -rf $__TMP
+        unzip -qo $BACKUP_HOME/$BACKUP_NAME.zip -d $__TMP
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+backup() {
+    __confirm $SOURCE_NS
+    mkdir -p $__TMP/resources $__TMP/data
+    oc project $SOURCE_NS
+
+    # export operator version
+    local op_pod="$(oc get pods | grep strimzi-cluster-operator | grep Running | cut -d " " -f1)"
+    oc exec -it $op_pod -- env | grep "STRIMZI_VERSION" > $__TMP/README
+
+    # export resources
+    __export "kafkas"
+    __export "kafkatopics"
+    __export "kafkausers"
+    __export "kafkabridges"
+    __export "kafkaconnectors"
+    __export "kafkaconnects"
+    __export "kafkaconnects2is"
+    __export "kafkamirrormaker2s"
+    __export "kafkamirrormakers"
+    __export "kafkarebalances"
+    # internal certificates and user secrets
+    __export "secrets" "strimzi.io/name=strimzi"
+    # custom configmap and secrets
+    for name in "${CUSTOM_CMS[@]}"; do
+        __export "cm/$name"
+    done
+    for name in "${CUSTOM_SECRETS[@]}"; do
+        __export "secret/$name"
+    done
+
+    # stop operator and statefulsets
+    oc scale deployment strimzi-cluster-operator --replicas 0
```

Comment:
Scaling down the operator is too disruptive since it affects also other clusters. Thsi should make use of the Pause feature which is being worked on in https://github.com/strimzi/strimzi-kafka-operator/pull/4399

Reply:
That's interesting, thanks.